build/Helix/ProcessHelixFiles.ps1

```diff
@@ -20,13 +20,31 @@ function Generate-File-Links
         Out-File -FilePath $helixLinkFile -Append -InputObject "<ul>"
         foreach($file in $files)
         {
-            Out-File -FilePath $helixLinkFile -Append -InputObject "<li><a href=$($file.Link)>$($file.Name)</a></li>"
+            $url = Append-HelixAccessTokenToUrl $file.Link "{Your-Helix-Access-Token-Here}"
+            Out-File -FilePath $helixLinkFile -Append -InputObject "<li>$($url)</li>"
         }
         Out-File -FilePath $helixLinkFile -Append -InputObject "</ul>"
         Out-File -FilePath $helixLinkFile -Append -InputObject "</div>"
     }
 }
 
+function Append-HelixAccessTokenToUrl
+{
+    Param ([string]$url, [string]$token)
+    if($token)
+    {
+        if($url.Contains("?"))
+        {
+            $url = "$($url)&access_token=$($token)"
+        }
+        else
+        {
+            $url = "$($url)?access_token=$($token)"
+        }
+    }
```

Comment:
i know you copied this from MUX but _i really wonder why they didn't use the system.url functions to do this_

Reply:
```powershell
function Append-QueryParameter {
    param (
        [System.UriBuilder]$Uri,
        [hashtable]$Parameters
    )

    $params = [System.Web.HttpUtility]::ParseQueryString($Uri.Query);
    foreach ($h in $Parameters.GetEnumerator()) {
        $params.Set($h.Name, $h.Value)
    }
    $uri.Query = $params.ToString()

    return $uri.Uri.ToString()
}
```

```powershell
> Append-QueryParameter "http://www.google.com/?foo=bar" @{"a" = "b"; "1" = "2"}
"http://www.google.com/?foo=bar&a=b&1=2"
```

👍